qualtran/bloqs/block_encoding/phase.py

```diff
@@ -0,0 +1,122 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from functools import cached_property
+from typing import Dict, Set, Tuple
+
+from attrs import frozen
+
+from qualtran import bloq_example, BloqBuilder, BloqDocSpec, QAny, Register, Signature, SoquetT
+from qualtran.bloqs.basic_gates import GlobalPhase
+from qualtran.bloqs.block_encoding import BlockEncoding
+from qualtran.bloqs.block_encoding.lcu_select_and_prepare import PrepareOracle
+from qualtran.resource_counting import BloqCountT, SympySymbolAllocator
+from qualtran.symbolics import SymbolicFloat, SymbolicInt
+
+
+@frozen
+class Phase(BlockEncoding):
+    r"""Apply a phase to a block encoding.
+
+    Given $B[A]$ as a $(\alpha, a, \epsilon)$-block encoding of $A$, produces a
+    $(\alpha, a, \epsilon)$-block encoding $B[\exp(i\pi\phi)A]$ of $\exp(i\pi\phi)A$.
```

Comment:
nit / feel free to ignore: "produces a block encoding $B[x]$ of $x$." took me some mental energy to parse that $x$ was indeed the same in the two expressions and you were just being precise about the notation. Maybe just "produces a block encoding of $x$"?